source/codegen/metadata/nidaqmx/enums.py

```diff
@@ -1245,81 +1276,13187 @@
             },
             {
                 'documentation': {
-                    'description': 'Direct Current.'
+                    'description': 'Direct Current.'
+                },
+                'name': 'DC',
+                'value': 10050
+            }
+        ]
+    },
+    'CurrentShuntResistorLocation1': {
+        'values': [
+            {
+                'documentation': {
+                    'description': 'Use the built-in shunt resistor of the device.'
+                },
+                'name': 'INTERNAL',
+                'value': 10200
+            },
+            {
+                'documentation': {
+                    'description': ' Use a shunt resistor external to the device. You must specify the value of the  shunt resistor by using DAQmx_AI_CurrentShunt_Resistance.'
+                },
+                'name': 'EXTERNAL',
+                'value': 10167
+            }
+        ]
+    },
+    'CurrentShuntResistorLocationWithDefault': {
+        'values': [
+            {
+                'name': 'DEFAULT',
+                'value': -1
+            },
+            {
+                'documentation': {
+                    'description': 'Internal'
+                },
+                'name': 'INTERNAL',
+                'value': 10200
+            },
+            {
+                'documentation': {
+                    'description': 'External'
+                },
+                'name': 'EXTERNAL',
+                'value': 10167
+            }
+        ]
+    },
+    'CurrentUnits1': {
+        'values': [
+            {
+                'documentation': {
+                    'description': 'Amperes.'
+                },
+                'name': 'AMPS',
+                'value': 10342
+            },
+            {
+                'documentation': {
+                    'description': ' Units a custom scale specifies. If you select this value, you must specify a  custom scale name.'
+                },
+                'name': 'FROM_CUSTOM_SCALE',
+                'value': 10065
+            },
+            {
+                'documentation': {
+                    'description': ' Units defined by TEDS information associated with the channel.'
+                },
+                'name': 'FROM_TEDS',
+                'value': 12516
+            }
+        ]
+    },
+    'CurrentUnits2': {
+        'values': [
+            {
+                'name': 'AMPS',
+                'value': 10342
+            },
+            {
+                'name': 'FROM_CUSTOM_SCALE',
+                'value': 10065
+            }
+        ]
+    },
+    'DAQmxErrors': {
+        'force-include': True,
```

Comment:
Where did `force-include` come from? I don't see it in the scrapigen main branch or in https://github.com/ni/grpc-device-scrapigen/pull/136

Reply:
If I copy the metadata from the scrapigen trunk, `force-include` is removed from 4 enums:

```
diff --git a/source/codegen/metadata/nidaqmx/enums.py b/source/codegen/metadata/nidaqmx/enums.py
index 0961b121..e7516ea5 100644
--- a/source/codegen/metadata/nidaqmx/enums.py
+++ b/source/codegen/metadata/nidaqmx/enums.py
@@ -1361,7 +1361,6 @@ enums = {
         ]
     },
     'DAQmxErrors': {
-        'force-include': True,
         'values': [
             {
                 'documentation': {
@@ -13539,7 +13538,6 @@ enums = {
         ]
     },
     'DAQmxWarnings': {
-        'force-include': True,
         'values': [
             {
                 'documentation': {
@@ -15926,7 +15924,6 @@ enums = {
         ]
     },
     'PowerIdleOutputBehavior': {
-        'force-include': True,
         'values': [
             {
                 'documentation': {
@@ -15945,7 +15942,6 @@ enums = {
         ]
     },
     'PowerOutputState': {
-        'force-include': True,
         'values': [
             {
                 'documentation': {
@@ -18480,4 +18476,4 @@ enums = {
             }
         ]
     }
-}
\ No newline at end of file
+}
```

Reply:
I have added `force-include` based on the comment provided by @reckenro here. https://github.com/ni/grpc-device-scrapigen/pull/112#discussion_r1116023009.

I have updated the code to latest and here I can see only `DAQmxErrors` and `DAQmxWarnings` need to be handled. So, I think we need a special case in the grpc-device codegen to exclude these.

Reply:
I have added `force-include` based on the comment provided by @reckenro here. https://github.com/ni/grpc-device-scrapigen/pull/112#discussion_r1116023009.

I have updated the code to latest and here I can see only `DAQmxErrors` and `DAQmxWarnings` need to be handled. So, I think we need a special case in the grpc-device codegen to exclude these.

Reply:
Oh yes, what Brad said is correct. I should've clarified a bit in that original comment.

If we wanted these enums to be in the .proto file we'd have to either reference them in functions.py/attributes.py or mark the enums 'force-include'.

We don't want to do that, though, so in grpc-device we'll have to either ease up the metadata_validation to not care about enums that are "unused" or add a validation suppression for these specific enums so it skips over this check just for them. I think the latter makes more sense because in general it's nice for most of our drivers to notice enums in enums.py that aren't used and thus not included in the .proto file.

For the suppression, it'll be similar to the one we have for [not allowing duplicate enum values](https://github.com/ni/grpc-device/blob/ad35fedd1975a3efbd84ffda3d3f2934377f0e9a/source/codegen/metadata_validation.py#L359) and that's marked up in the driver's enums_addons.py like [here](https://github.com/ni/grpc-device/blob/0161a4ccedbdba6aa0f53b8f87308a3805a5f2ff/source/codegen/metadata/nixnet/enums_addon.py#L7). You should add a new rule for ENUMS_SHOULD_BE_USED or something and mark up the enums_addon.py to note this suppression for the enums and then we'll skip over throwing the exception [here](https://github.com/ni/grpc-device/blob/ad35fedd1975a3efbd84ffda3d3f2934377f0e9a/source/codegen/metadata_validation.py#L363) if the rule is set for the enum.

Reply:
Thanks @bkeryan and @reckenro for the suggestions. Sorry that I misunderstood the conversation and added `force-include` property before.

Now, I have removed the same and added a validation suppression rule to exclude these.

https://github.com/ni/grpc-device/pull/881/commits/b23fc4a4e0cdcaaf477b37b8eebe1af046b5df87